### DISCUSSION TITLE
Don't swallow Token-Validation Errors

### DISCUSSION THREAD

**Discussion by andymel123** [keycloak/keycloak#15483] [Q&A]
I stepped through the token validation code of keycloak-connect today as I had a `403 - access denied` issue and had no clue why.

There are Error objects returned in the [validateToken function](https://github.com/keycloak/keycloak-nodejs-connect/blob/main/middleware/auth-utils/grant-manager.js#L415) that would provide valuable information about the validation problem but it seems like they are getting swallowed by a [catch without error parameter in keycloak.js](https://github.com/keycloak/keycloak-nodejs-connect/blob/main/keycloak.js#L330)

Is this on purpose?
Is there a way to retrieve the error messages provided by validateToken that I don't get?

Is it better to create an issue in the keycloak/keycloak-nodejs-connect repo or write to the keycloak-dev mailing list?

**Comment by LakshmiPhani7680** [comment]
I am also getting 403 Error Even after login for specific application, remaining apps not getting it
The only error is Could Not obtain Grant Code, can anyone help with this? @andymel123
